### PR TITLE
feat: Add override methods and docs

### PR DESCRIPTION
**Pull request by MichaelUnkey**
## What does this PR do?

Fixes # (issue)

*If there is not an issue for this, please create one first. This is used to tracking purposes and also helps use understand why this PR exists*

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] Enhancement (small improvements)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

Docs match routes and make sense. 

### Required

- [x] Filled out the "How to test" section in this PR
- [x] Read [Contributing Guide](./CONTRIBUTING.md)
- [x] Self-reviewed my own code
- [x] Commented on my code in hard-to-understand areas
- [x] Ran `pnpm build`
- [x] Ran `pnpm fmt`
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the Unkey Docs if changes were necessary

## Summary by CodeRabbit

- **New Features**
	- Added comprehensive documentation for the following functionalities within the SDK:
		- Delete Override
		- Get Override
		- List Overrides
		- Set Override
	- Introduced a new group titled "Overrides" in the documentation structure.
	- Expanded public API to include new methods for managing rate limit overrides.

- **Documentation**
	- Enhanced organization and clarity of documentation related to rate limit overrides.
	- Updated documentation for the TypeScript client with new configuration options.

- **Chores**
	- Minor adjustments to the formatting and structure of configuration files for consistency.

### CODE REVIEW

File: packages/ratelimit/src/overrides.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  }
+  public get setOverride() {
+    return this.unkey.ratelimits.setOverride;
+  }
+  public get deleteOverride() {
+    return this.unkey.ratelimits.deleteOverride;
+  }
+  public get listOverrides() {
+    return this.unkey.ratelimits.listOverrides;
+  }
```
[the remark this answers is not in the document]
Comment (chronark) [reply packages/ratelimit/src/overrides.ts:50]: this was correct, the getter method itself is not asynchronous
it's a synchronosu function that returns an asynchronous function
so we need to remove the a`async`

[the remark this answers is not in the document]
Comment (chronark) [reply packages/ratelimit/src/overrides.ts:50]: @MichaelUnkey

**Review by chronark — CHANGES_REQUESTED** [review]

File: packages/ratelimit/src/overrides.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (MichaelUnkey) [reply packages/ratelimit/src/overrides.ts:50]: This was remnant from our conversation yesterday. Thought it was reverted back but I guess not everything stuck.  @chronark if I understand your comments correctly, ignore this part of the rabbit comment "typeof this.unkey.ratelimits.setOverride {"   ??

File: apps/docs/libraries/ts/sdk/ratelimits/overrides/delete-override.mdx [hunk 2]
```diff
@@ -0,0 +1,41 @@
+---
+title: "Delete Override"
+description: "Deletes an override"
+---
+
+## Request
+<ParamField body="identifier" type="string" required>
+Identifier of your user, this can be their userId, an email, an ip or anything else. Wildcards ( * ) can be used to match multiple identifiers, More info can be found at https://www.unkey.com/docs/ratelimiting/overrides#wildcard-rules
```
[the remark this answers is not in the document]
Comment (MichaelUnkey) [reply apps/docs/libraries/ts/sdk/ratelimits/overrides/delete-override.mdx:8]: @chronark Ignore this. It seems wrong to me?

[the remark this answers is not in the document]
Comment (chronark) [reply apps/docs/libraries/ts/sdk/ratelimits/overrides/delete-override.mdx:8]: the link works, lgtm 👍

File: apps/docs/libraries/ts/sdk/ratelimits/overrides/delete-override.mdx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+## Response
+
+No response, but if no error is returned the override has been deleted successfully.
+
+
+<RequestExample>
+
+```ts
+const override = await unkey.ratelimit.deleteOverride({
```
Comment (chronark) [inline apps/docs/libraries/ts/sdk/ratelimits/overrides/delete-override.mdx:32]: didn't we just talk about this in DMs?
for deletes we don't want to assign a variable

File: apps/docs/libraries/ts/sdk/ratelimits/overrides/delete-override.mdx [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    identifier: "user_123",
+    namespaceName: "email.outbound",
+})
+```
+```ts
+const override = await unkey.ratelimit.deleteOverride({
+    identifier: "user_123",
+    namespaceId: "rlns_1234",
+})
+```
```
[the remark this answers is not in the document]
Comment (chronark) [reply apps/docs/libraries/ts/sdk/ratelimits/overrides/delete-override.mdx:42]: yes

File: apps/docs/libraries/ts/sdk/ratelimits/overrides/list-overrides.mdx [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+</ResponseField>
+ </Expandable>
+
+</ResponseField>
+
+
+<RequestExample>
+  
+```ts
+const override = await unkey.ratelimit.listOverrides({
```
Comment (chronark) [inline apps/docs/libraries/ts/sdk/ratelimits/overrides/list-overrides.mdx:65]: plural please

File: apps/docs/libraries/ts/sdk/ratelimits/overrides/list-overrides.mdx [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+    nameSpaceId:"rlns_12345", 
+});
+```
+```ts
+const override = await unkey.ratelimit.listOverrides({
```
Comment (chronark) [inline apps/docs/libraries/ts/sdk/ratelimits/overrides/list-overrides.mdx:70]: plural please

File: apps/docs/libraries/ts/sdk/ratelimits/overrides/list-overrides.mdx (hunk 5 above)
Comment (chronark) [reply apps/docs/libraries/ts/sdk/ratelimits/overrides/list-overrides.mdx:65]: I don't understand this tbh, I just marked this and even gave you a committable example
CleanShot 2024-12-12 at 19 42 26@2x

**Review by chronark — CHANGES_REQUESTED** [review]

**Review by chronark — APPROVED** [review]

**Pull request merged**